### PR TITLE
Cg/friends dash

### PR DESCRIPTION
**Pull request by GOECHA**
# Pull Request

#Display Friend Events

- [x] New feature (adds functionality)
- Describe the purpose of this PR(other):

  - Display friends on the FriendsDashboard

# Please describe the tests that you ran to verify your changes.

- [x] I tested my changes in the browser
- [ ] I tested with Cypress

Display Friend's events on the FriendsDashboard

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new bugs
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any other comments, questions or concerns?
- [ ] I have added screenshots,  (see below):

### CODE REVIEW

**Review by abbyluce — APPROVED** [review]
Yayyyy nice work!

File: src/components/App/App.js [hunk 1]
```diff
@@ -28,9 +58,29 @@ const App = () => {
...
+    let matches = []
+    data.users.forEach((friend) => {
+    currentUser.friends.forEach((friend2) => {
+      if (friend.name === friend2.name) {
+        matches.push(friend.events)
+      }
+    });
+  });
+
+  setCurrentUser({...currentUser, friendEvents: matches.flat()});
```
Comment (nicktu12) [inline src/components/App/App.js:73]: nice use of `flat`

File: src/components/EventCard/EventCard.js [hunk 2]
```diff
@@ -6,9 +6,17 @@ const EventCard = ({ name, date, venue, city, state, image, id }) => {
   const [year, month, day] = date.split("-");
   const dateDisplay =
     month && day && year ? `${month}/${day}/${year}` : `Multi-day Event`;
-
+console.log(`&&&`, image);
   return (
-    <div className="event-card">
+    <div className="event-card"
+      style={{
```
Comment (nicktu12) [inline src/components/EventCard/EventCard.js:12]: Curious why there are inline styles here?

File: src/components/SearchForm/SearchForm.js [hunk 3]
```diff
@@ -1,51 +1,60 @@
...
+  const [message, setMessage] = useState("Search for an upcoming show!");
+
+  const handleChange = (e) => {
+    setArtistName(e.target.value);
+  };
+
+  const handleClick = (e) => {
+    e.preventDefault();
+    artistName
+      ? fetch(
```
Comment (nicktu12) [inline src/components/SearchForm/SearchForm.js:17]: I'd like to see fetch calls abstracted to a utility file, or at least another method in this component - can be hard to reach when the fetch is conditionally firing based on a ternary value

File: src/components/SearchForm/SearchForm.js [hunk 4]
```diff
@@ -1,51 +1,60 @@
...
+          .then((data) => {
+            console.log(`data.data.events`, data.errors);
+            if (!data.errors) {
+              setResults(data.data.events);
+              setSearchedEvents(data.data.events);
+              setMessage("");
+            } else {
+              setMessage("No events found . . .");
+            }
+          })
```
Comment (nicktu12) [inline src/components/SearchForm/SearchForm.js:30]: Add `.catch()`

File: src/hooks/useUser.js [hunk 5]
```diff
@@ -6,6 +6,10 @@ const GET_USER = gql`
       name
       id
       image
+      friends {
```
Comment (nicktu12) [inline src/hooks/useUser.js:9]: nice gQL

File: src/components/EventCard/EventCard.js (hunk 2 above)
Comment (abbyluce) [reply src/components/EventCard/EventCard.js:12]: This was the best way we found to make the background image dynamic and speak to each event's image.

File: src/components/SearchForm/SearchForm.js (hunk 4 above)
Comment (jrmedina) [reply src/components/SearchForm/SearchForm.js:30]: Added!

File: src/components/SearchForm/SearchForm.js (hunk 3 above)
Comment (jrmedina) [reply src/components/SearchForm/SearchForm.js:17]: Done!

File: src/components/App/App.js [hunk 6]
```diff
@@ -28,9 +58,29 @@ const App = () => {
     return foundEvent;
   };
 
+
+
+  const findFriends = () => {
+    let matches = []
+    data.users.forEach((friend) => {
+    currentUser.friends.forEach((friend2) => {
```
[the remark this answers is not in the document]
Comment (jrmedina) [reply src/components/App/App.js:66]: Looking into other solutions...

**Pull request merged**